misc/acrn-config/config_app/controller.py

```diff
@@ -1,288 +1,291 @@
-        :param args: the path of the element.
-        :return: None.
-        """
-        if self._curr_xml_tree is None:
-            return
-        dest_node = self._get_dest_node(*args)
-        self._curr_xml_tree.getroot().remove(dest_node)
-
-    def _get_dest_node(self, *args):
-        """
-        get the destination element by its path.
-        :param args: the path of the element.
-        :return: the destination element.
-        """
-        if self._curr_xml_tree is None:
-            return None
-        dest_node = self._curr_xml_tree.getroot()
-        path = '.'
-        for arg in args:
-            # tag:attr=xxx
-            # tag:attr
-            # tag
-            tag = None
-            attr_name = None
-            attr_value = None
-            if ':' not in arg:
-                tag = arg
-            elif '=' not in arg:
-                # tag = arg.split(':')[0]
-                # attr_name = arg.split(':')[1]
-                raise Exception('unsupported xml path: tag:attr')
-            else:
-                tag = arg.split(':')[0]
-                attr = arg.split(':')[1]
-                attr_name = attr.split('=')[0]
-                attr_value = attr.split('=')[1]
-
-            if attr_value is None:
-                path += ("/" + tag)
-            else:
-                path += ("/" + tag + "[@" + attr_name + "='" + attr_value + "']")
-
-        dest_node = dest_node.findall(path)
-        if dest_node is not None and dest_node != []:
-            return dest_node[0]
-
-        raise Exception('can not find node by {} from xml'.format(args))
-
-    def save(self, xml=None):
-        """
-        save current xml to file.
-        :param xml: the file name to save; if not specified, save current xml to default names.
-        :return: None.
-        """
-        if self._curr_xml_tree is None:
-            return
-        if xml is None:
-            xml = self._curr_xml
-
-        xml_path = os.path.join(self._xml_path, 'user_defined')
-        if not os.path.isdir(xml_path):
-            os.makedirs(xml_path)
-
-        self._format_xml(self._curr_xml_tree.getroot())
-        self._curr_xml_tree.write(os.path.join(xml_path, xml+'.xml'), encoding='utf-8',
-                                  xml_declaration=True, method='xml')
-
-    def _format_xml(self, element, depth=0):
-        i = "\n" + depth * "    "
-        if element:
-            if not element.text or not element.text.strip():
-                element.text = i + "    "
-            if not element.tail or not element.tail.strip():
-                element.tail = i
-            for element in element:
-                self._format_xml(element, depth + 1)
-            if not element.tail or not element.tail.strip():
-                element.tail = i
-        else:
-            if depth and (not element.tail or not element.tail.strip()):
-                element.tail = i
+# Copyright (C) 2019 Intel Corporation.
```

Comment:
Why there are diffs for the parts that are not modifed?
It is difficult to review the modified part...

Reply:
@binbinwu1 there are 2 patches for this PR. The first one is: acrn-config: change source file format from dos(line ending as CRLF) to unix format(line ending as LF). You can find the diffs from maillist.